src/catchup/CatchupConfiguration.cpp

```diff
@@ -16,6 +16,18 @@ CatchupConfiguration::CatchupConfiguration(LedgerNumHashPair ledgerHashPair,
 {
 }
 
+CatchupConfiguration::CatchupConfiguration(HistoryArchiveState has,
+                                           LedgerHeaderHistoryEntry lhhe)
+    : mCount(0)
+    , mLedgerHashPair(LedgerNumHashPair(lhhe.header.ledgerSeq,
+                                        std::make_optional(lhhe.hash)))
+    , mMode(CatchupConfiguration::Mode::LOCAL)
+    , mHAS(std::make_optional(has))
+    , mHistoryEntry(std::make_optional(lhhe))
+{
+    releaseAssert(mHAS->currentLedger == mHistoryEntry->header.ledgerSeq);
+}
+
```

Comment:
done. All constructors now call `checkInvariants` function, which checks several things. This also revealed a bug in ` CatchupConfiguration::resolve`, which I did not update correctly to handle the local mode (it was not caught by tests because the result of `resolve` is only used to calculate catchup range, so it doesn't use the new fields)